tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,1311 @@
+            StartTag stag;
+            EndTag etag;
+            // Track the node level nesting of nodes handled in this function only
+            int level = 1;
+
+            // To preserve order, the expected format for transforms is:
+            // <Transform>
+            //     <![CDATA[...XSLT...]]>
+            // </Transform>
+            appendStartTag(xpp, root, false, true);
+            appendStartCDATA();
+
+            do
+            {
+                type = xpp->next();
+                switch (type)
+                {
+                    case XmlPullParser::START_TAG:
+                        xpp->readStartTag(stag);
+                        if (isManifestNodeNamed(stag, MANIFEST_TAG_INCLUDE))
+                            processInclude(xpp, stag, IncludeType::xslt);
+                        else
+                        {
+                            appendStartTag(xpp, stag);
+                            level++;
+                        }
+                        break;
+
+                    case XmlPullParser::CONTENT:
+                        appendNonWhitespaceContent(xpp);
+                        break;
+
+                    case XmlPullParser::END_TAG:
+                        xpp->readEndTag(etag);
+
+                        // We're about to encounter the root <Transform> node, so close
+                        // out the CDATA section we injected first
+                        if (isRootEndTagNamed(etag, "Transform", level))
+                        {
+                            appendEndCDATA();
+                            appendEndTag(etag, false);
+                        }
+                        else
+                            appendEndTag(etag);
+
+                        level--;
+                        break;
+                }
+                lastType = type;
+            }
+            while (level > 0);
+
+            return type;
+        }
+
+        /**
+         * @brief Process the 'EsdlDefinition' node and children.
+         *
+         * The 'EsdlDefinition' node is renamed on output to 'Definitions'. Its contents are enclosed
+         * with CDATA.
+         *
+         * @param xpp pointer to the pull parser
+         * @param root reference to the 'EsdlDefinition' node read
+         * @return int type of the last node read
+         */
+        int processEsdlDefinition(IXmlPullParser* xpp, StartTag& root)
+        {
+            int type;
+            StartTag stag;
+            EndTag etag;
+            int level = 1;
+
+            if (foundEsdlDefn)
+                throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Only one EsdlDefinition per manifest is supported. Found a second at line #%d, column #%d", xpp->getLineNumber(), xpp->getColumnNumber());
+            foundEsdlDefn = true;
+
+            if (outputType != ManifestType::bundle)
+            {
+                UWARNLOG("The requested binding output will not include the ESDL Definition found in the Manifest.");
```

Comment:
For the second case what is the operative difference from the first? Is it the difference between using the `@outputType=binding` attribute instead of passing it on the CLI?

Reply:
In the first case, the manifest is declared to be capable of producing a bundle. It can explicitly also be used to produce a binding, which is a subset of a bundle. I don't see a need to warn the user that the user has explicitly requested a proper subset of the template.

In the second case, the manifest is explicitly declared to produce a binding. There should be no expectation that an ESDL definition is present and a valid bundle can be output. If a definition is present, will its inclusion in binding output be valid? Should a definition be silently ignored in binding output? Should a user be allowed to request a bundle from a binding template? These are conditions that may warrant generating warning or error messages.

I was not, and am not, suggesting that alternate messaging is necessary now. I was just making the point that there are other, more worthy, candidates for warning and error messages than this.